### DISCUSSION TITLE
How to get the default Sql DBType of a property, or from Column Attribute?

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [linq2db/linq2db#4614] [Q&A]
I am creating a tool to automatically update my DB Schema, i.e., add new columns if needed.
But the DbType is always empty from the ColumnAttribute. I thought there should be a way to get it, since LinqToDB must already know it.

**Comment by MaceWindu** [comment]
`ColumnAttribute.DbType` provides type name for Linq To DB, not vice versa. Getting actual type will require you to use sql builder API.

Approximate code will be:

```cs
var sqlBuilder = dataProvider.CreateSqlBuilder(...);
foreach (var column in ms.GetEntityDescriptor(typeof(Entity)).Columns)
{
    var typeName = new StringBuilder();
    sqlBuilder.BuildDataType(typeName, column.GetDbDataType(true));
}
```

But this API could return wrong type names for MySQL as it doesn't accept `bool forCreateTable` flag we use internally to generate type name when it differ for create table and e.g. type cast expressions.

There is no plans to support better api.
